eth/stagedsync/bor_heimdall_shared.go

```diff
@@ -124,6 +124,9 @@ func fetchRequiredHeimdallSpansIfNeeded(
 	if bor.IsBlockInLastSprintOfSpan(toBlockNum, cfg.borConfig) {
 		requiredSpanID++
 	}
+	if bor.IsFirstSprintStart(toBlockNum, cfg.borConfig) {
```

Comment:
I would suggest to use an else if here and also include `spanID==0 && IsFirstSprintStart` to make it clear to the reader that this is a case that only happens in spanID=0 

also please leave a comment explaining why - this is a bit of tribal knowledge that is not clear otherwise

would be great if you can also add a unit test for this scenario in `stage_bor_heimdall_test.go` to avoid regressions

Reply:
Sure, so as mentioned in the comment above, span 1 (and indirectly span 0) are committed into genesis contracts in the very first sprint. As per the logic of `fetchRequiredHeimdallSpansIfNeeded` function, it explicitly handles fetching of span 0 and then for rest of the spans, it fetches it at the start of last sprint of current span. But, for span 1, as it's committed a bit early, we'd need the span to be fetched from heimdall early else it will result in missing span in bor consensus leading to liveness issue.

Re. tests, sure. Will try to add them. Thanks!